addons/binding/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/internal/discovery/NeeoBrainDiscovery.java

```diff
@@ -0,0 +1,139 @@
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.io.transport.mdns.discovery.MDNSDiscoveryParticipant;
+import org.openhab.binding.neeo.NeeoBindingConstants;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Implementation of {@link MDNSDiscoveryParticipant} that will discover NEEO brain(s).
+ *
+ * @author Tim Roberts - initial contribution
+ */
+@Component(service = MDNSDiscoveryParticipant.class, immediate = true, name = "org.openhab.binding.neeo.discovery.NeeoBrainDiscovery")
+public class NeeoBrainDiscovery implements MDNSDiscoveryParticipant {
+
+    /** The logger. */
+    private Logger logger = LoggerFactory.getLogger(NeeoBrainDiscovery.class);
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see org.eclipse.smarthome.io.transport.mdns.discovery.MDNSDiscoveryParticipant#getSupportedThingTypeUIDs()
+     */
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return Collections.singleton(BRIDGE_TYPE_BRAIN);
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see org.eclipse.smarthome.io.transport.mdns.discovery.MDNSDiscoveryParticipant#getServiceType()
+     */
+    @Override
+    public String getServiceType() {
+        return "_neeo._tcp.local.";
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see
+     * org.eclipse.smarthome.io.transport.mdns.discovery.MDNSDiscoveryParticipant#createResult(javax.jmdns.ServiceInfo)
+     */
+    @Override
+    public DiscoveryResult createResult(ServiceInfo service) {
+        ThingUID uid = getThingUID(service);
+        if (uid == null) {
+            return null;
+        }
+
+        final Map<String, Object> properties = new HashMap<>(2);
+
+        final InetAddress ip = getIpAddress(service);
+        if (ip == null) {
+            return null;
+        }
+        final String inetAddress = ip.toString().substring(1); // trim leading slash
+
+        final String id = uid.getId();
+        final String label = service.getName() + " (" + id + ")";
+
+        properties.put(NeeoBindingConstants.CONFIG_IPADDRESS, inetAddress);
+
+        logger.debug("Adding NEEO Brain to inbox: {} at {}", id, inetAddress);
+        return DiscoveryResultBuilder.create(uid).withProperties(properties).withLabel(label).build();
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see
+     * org.eclipse.smarthome.io.transport.mdns.discovery.MDNSDiscoveryParticipant#getThingUID(javax.jmdns.ServiceInfo)
+     */
+    @Override
+    public ThingUID getThingUID(ServiceInfo service) {
+        if (!StringUtils.equals("neeo", service.getApplication())) {
+            return null;
+        }
+
+        if (getIpAddress(service) == null) {
+            return null;
+        }
+
+        final String model = service.getPropertyString("hon"); // model
+        if (model == null || !model.toLowerCase().startsWith("neeo") && model.length() > 5) {
```

Comment:
I'm assuming you want to state if _does_ start with neeo, but length <= 5 it should return null. But it states if not start with neeo, but length <= 5 it does not return null. It should probably be: or last: `|| model.length <= 5` or `!(model.toLowerCase().startsWith("neeo") && model.length() > 5)`